libs/@blockprotocol/graph/src/stdlib/interval.ts

```diff
@@ -0,0 +1,333 @@
+import {
+  NonNullTimeInterval,
+  TemporalBound,
+  Timestamp,
+  TimestampLimitedTemporalBound,
+} from "../types/temporal-versioning.js";
+import { boundIsAdjacentTo, compareBounds } from "./bound.js";
+
+/**
+ * Checks whether two given {@link NonNullTimeInterval}s are adjacent to one another, where adjacency is defined as
+ * being next to one another on the timeline, without any points between, *and where they are not overlapping*. Thus,
+ * if adjacent, the two intervals should span another given interval.
+ *
+ * @param lhs - The first interval of the comparison (order is unimportant)
+ * @param rhs - The second interval of the comparison
+ */
+export const intervalIsAdjacentToInterval = (
+  lhs: NonNullTimeInterval,
+  rhs: NonNullTimeInterval,
+): boolean => {
+  /*
```

Comment:
I did try adding these comments to the actual doc-strings, but none of the formatting ended up being readable so I opted to put it back down in a normal in-code comment like we've done in HASH. The problems with the formatting included the table rendering removing whitespaces (which is essential), weirdly formatting the intervals, etc.

![image](https://user-images.githubusercontent.com/25749103/215833858-bd57493d-de8e-4a96-819a-04c6dda599dd.png)
![image](https://user-images.githubusercontent.com/25749103/215833880-3c9688c6-2327-4845-b45e-2039a5c24ee8.png)
![image](https://user-images.githubusercontent.com/25749103/215833898-b5908ef8-b353-4cf5-9460-6b8631264188.png)